### PR TITLE
Introduce sphinx processing for html API web site

### PR DESCRIPTION
**Pull request by jwhonce**
* Add docs target to Makefile
* Started updating some docstrings
* Added support for connections from containers.conf
* Ported more uses of XDG_* envvar to xdg package
* Downgrade modules in requirements.txt as needed by RHEL/Fedora
  packaging.

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by jwhonce** [comment]
@jnovy I believe these package versions will work for you. See [requirements.txt](https://github.com/containers/podman-py/pull/85/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552R3). /cc @lsm5

File: docs/source/conf.py [hunk 1]
```diff
@@ -48,7 +50,8 @@
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
 #
-html_theme = 'alabaster'
+# html_theme = 'alabaster'
+html_theme = 'default'
 html_theme_options = {
     'description': 'A Python library for the Podman service API',
     'fixed_sidebar': True,
```
Comment (mwhahaha) [inline docs/source/conf.py:57]: JFYI

WARNING: unsupported theme option 'description' given
WARNING: unsupported theme option 'fixed_sidebar' given

**Review by mwhahaha — APPROVED** [review]
looks ok

File: docs/source/index.rst [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
     :inherited-members:
 
 .. toctree::
-   :maxdepth: 2
-   :caption: Contents:
-
+    :maxdepth: 2
+    :caption: Contents:
 
+   modules
```
[the remark this answers is not in the document]
Comment (jwhonce) [reply docs/source/index.rst:21]: Agreed, I'm still learning how sphinx works while keeping the packaging people unblocked. :-)

File: docs/source/conf.py (hunk 1 above)
Comment (jwhonce) [reply docs/source/conf.py:57]: Do you have a strong opinion on which theme is used?

Comment (jwhonce) [reply docs/source/conf.py:57]: And can you merge please. TIA

**Comment by lsm5** [comment]
Thanks @jwhonce . I'll create scratch builds for @jnovy to test

File: docs/source/conf.py (hunk 1 above)
Comment (mwhahaha) [reply docs/source/conf.py:57]: I really don't have an opinion at this time, just noting the warnings. having some docs are better than nothing. i'll merge this and we can play around with it later

**Comment by jwhonce** [comment]
@mwhahaha Thanks for the quick turn around!

**Comment by lsm5** [comment]
@jwhonce could you cut a new release with this please?

**Comment by lsm5** [comment]
hmm, setup.cfg still says urllib3 1.26.4. Intentional?

**Comment by h-vetinari** [comment]
> Downgrade modules in requirements.txt as needed by RHEL/Fedora packaging.

This sounds like it's not a hard requirement of the package but to help the RHEL packages - conda-forge does not have this limitation, so I'm not planning to lower the urllib pin...

**Pull request merged**